go/runtime/txpool/local_queue.go

```diff
@@ -69,6 +69,18 @@ func (lq *localQueue) HandleTxsUsed(hashes []hash.Hash) {
 	lq.txs = keptTxs
 }
 
+func (lq *localQueue) PeekAll() []*TxQueueMeta {
+	lq.l.Lock()
+	defer lq.l.Unlock()
+
+	allTxs := lq.txs
+	var txs []*TxQueueMeta
+	for _, tx := range allTxs {
+		txs = append(txs, &TxQueueMeta{raw: tx.Raw(), hash: tx.Hash(), firstSeen: tx.FirstSeen()})
```

Comment:
I thought it would be better to clone them, since they get returned to code outside of the txpool package and I don't know how the existing txpool code manipulates the structures, so returning pointers doesn't seem safe to me.

Reply:
All other methods return existing pointers and I don't see any places that would mutate them. They also don't contain any fields that could reasonably change (data, hash and first seen time).

Reply:
OK, I've changed it to just copy the pointers into a new slice :)